codec-http/src/main/java/io/netty/handler/codec/http/HttpMethod.java

```diff
@@ -87,18 +87,116 @@
     public static final HttpMethod CONNECT = new HttpMethod("CONNECT", true);
 
     private static final Map<String, HttpMethod> methodMap =
-            new HashMap<String, HttpMethod>();
+            new HashMap<String, HttpMethod>(9);
 
     static {
-        methodMap.put(OPTIONS.toString(), OPTIONS);
         methodMap.put(GET.toString(), GET);
-        methodMap.put(HEAD.toString(), HEAD);
-        methodMap.put(POST.toString(), POST);
         methodMap.put(PUT.toString(), PUT);
+        methodMap.put(POST.toString(), POST);
+        methodMap.put(HEAD.toString(), HEAD);
         methodMap.put(PATCH.toString(), PATCH);
-        methodMap.put(DELETE.toString(), DELETE);
         methodMap.put(TRACE.toString(), TRACE);
+        methodMap.put(DELETE.toString(), DELETE);
         methodMap.put(CONNECT.toString(), CONNECT);
+        methodMap.put(OPTIONS.toString(), OPTIONS);
+    }
+
+    /**
+     * Internal method which allows to create known {@link HttpMethod} without create a temporary {@link String}
+     * to reduce object creation and so GC-pressure.
+     */
+    static HttpMethod valueOf(char[] chars, int startIndex, int endIndex) {
+        startIndex = HttpDecoderUtil.startIndex(chars, startIndex, endIndex);
+        endIndex = HttpDecoderUtil.endIndex(chars, startIndex, endIndex);
+        int size = endIndex - startIndex;
+        if (size == 0) {
```

Comment:
Does it make sense to protect against negative (i.e. `if (size <= 0) { ... }`)?  If not consider adding javadoc comments to clarify assumptions.